src/main/java/trip/Participant.java

```diff
@@ -33,11 +37,17 @@ public void addTransportBooking(TransportBooking transport) {
         transportBookings.add(transport);
     }
 
-    public void addHotelBooking(HotelBooking hotel) {
-        if (hotelBookings == null) {
-            hotelBookings = new ArrayList<>();
+    public void addHotelBooking(HotelBooking hotelBooking) {
+        if(hotelBooking.calculatePrice() != null) {
+            if (hotelBookings == null) {
```

Comment:
hotelBooking.calculatePrice should fail if hotelBooking is null, so maybe you reverse the order of checks?

Reply:
In first IF I check if calculated price is null (in case of  lenght of staying in hotel is 0 or negative I set price to null)
In second IF I check if list of participant's hotel bookingS is null and if yes then I create new list and add the booking.